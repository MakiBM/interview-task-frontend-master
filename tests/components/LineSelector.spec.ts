import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import { createStore } from 'vuex'
import Component from '../../src/components/LineSelector.vue'

describe('LineSelector Component Snapshots', () => {
  it('matches the snapshot with no selected line', () => {
    const store = createStore({
      state: {
        selectedLine: null,
      },
      getters: {
        getLines: () => ['A', 'B', 'C'], // Mock lines
      },
      mutations: {
        SET_SELECTED_LINE: vi.fn(), // Mock mutation
      },
    })

    const wrapper = mount(Component, {
      global: {
        plugins: [store],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches the snapshot with a selected line', () => {
    const store = createStore({
      state: {
        selectedLine: 'A', // Mock a selected line
      },
      getters: {
        getLines: () => ['A', 'B', 'C'], // Mock lines
      },
      mutations: {
        SET_SELECTED_LINE: vi.fn(), // Mock mutation
      },
    })

    const wrapper = mount(Component, {
      global: {
        plugins: [store],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
