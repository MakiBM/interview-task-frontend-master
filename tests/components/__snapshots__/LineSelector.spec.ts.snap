// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LineSelector Component Snapshots > matches the snapshot with a selected line 1`] = `
"<div class="card border-0 p-2 front-inter" data-v-4b1ccc45="">
  <div class="card-body" data-v-4b1ccc45="">
    <p class="card-title mb-1 semibold" data-v-4b1ccc45=""> Select Bus Line </p>
    <div class="d-flex flex-wrap gap-2 pt-4" data-v-4b1ccc45=""><button class="btn btn-primary" style="background: rgb(41, 63, 108);" data-v-4b1ccc45="">A</button><button class="btn btn-primary" style="" data-v-4b1ccc45="">B</button><button class="btn btn-primary" style="" data-v-4b1ccc45="">C</button></div>
  </div>
</div>"
`;

exports[`LineSelector Component Snapshots > matches the snapshot with no selected line 1`] = `
"<div class="card border-0 p-2 front-inter" data-v-4b1ccc45="">
  <div class="card-body" data-v-4b1ccc45="">
    <p class="card-title mb-1 semibold" data-v-4b1ccc45=""> Select Bus Line </p>
    <div class="d-flex flex-wrap gap-2 pt-4" data-v-4b1ccc45=""><button class="btn btn-primary" style="" data-v-4b1ccc45="">A</button><button class="btn btn-primary" style="" data-v-4b1ccc45="">B</button><button class="btn btn-primary" style="" data-v-4b1ccc45="">C</button></div>
  </div>
</div>"
`;
